src/containers/Header/NetworkPicker/NetworkPicker.tsx

```diff
@@ -0,0 +1,131 @@
+import { useContext, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Dropdown, DropdownItem } from '../../shared/components/Dropdown'
+import SocketContext from '../../shared/SocketContext'
+import { ANALYTIC_TYPES, analytics } from '../../shared/utils'
+import './NetworkPicker.scss'
+
+export interface Network {
+  network: string
+  title: string
+  url: string
+}
+
+export const NetworkPicker = () => {
+  const CUSTOM_NETWORK_BASE_LINK = process.env.VITE_CUSTOMNETWORK_LINK
+  const STATIC_ENV_LINKS: Record<string, string | undefined> = {
+    mainnet: process.env.VITE_MAINNET_LINK,
+    testnet: process.env.VITE_TESTNET_LINK,
+    devnet: process.env.VITE_DEVNET_LINK,
+    amm: process.env.VITE_AMM_LINK,
+  }
+  const currentMode: string = process.env.VITE_ENVIRONMENT || 'mainnet'
+
+  const rippledSocket = useContext(SocketContext)
+  const { t } = useTranslation()
+  const [newRippledUrl, setNewRippledUrl] = useState('')
+
+  const rippledUrl = rippledSocket?.rippledUrl
+  const isCustom = currentMode === 'custom'
+  const inNetwork = !(isCustom && rippledUrl === undefined)
+
+  const getNetworkName = (network: string) =>
+    t(`network_name`, { context: network })
+
+  const getCustomNetworkName = (url) =>
+    `${getNetworkName('custom')}: ${url
+      ?.replace(`${CUSTOM_NETWORK_BASE_LINK || ''}/`, '')
+      .toLowerCase()}`
+
+  const trackNetworkSwitch = (network: Network) => {
+    analytics(ANALYTIC_TYPES.event, {
+      eventCategory: 'mode switch',
+      eventAction: network.url,
+    })
+  }
+
+  const networks = [
```

Comment:
Any reason not to make these things global?

Reply:
No but these are the only place these are used.

Reply:
I meant outside the render path. Global to the file.

Reply:
I moved it inside because it was messing with unit tests.  Having it inside the render made it happen later.

Is there a better way to mock environment variables for a test so that the file is evaluated during ever `it`?

Reply:
Used jest.isolateModules to test the env variables